packages/canvas/canvas-renderer/src/CanvasRenderer.js

```diff
@@ -247,10 +236,58 @@ export class CanvasRenderer extends AbstractRenderer
         context.restore();
 
         this.resolution = rootResolution;
+        this._projTransform = null;
 
         this.emit('postrender');
     }
 
+    /**
+     * sets matrix of context
+     * called only from render() methods
+     * takes care about resolution
+     * @param {PIXI.Matrix} transform world matrix of current element
```

Comment:
Could you make this private?

Reply:
setBlendMode isn't private

Reply:
It’s only used internally. No need to be public. Would like to limit the public API

Reply:
It shouldn't be private, as our WebGL state-related functions aren't private too.

Reply:
We expect from our users to write their own components, right? That override their own `_render` and `_renderCanvas` methods. We expose mid-level API. If we would mark mid-level API differently than high-level, I would be happy to go through all core and do it, but we dont have necessary tools. Same with low-level (internal, only for plugins).

Reply:
Of course there's an opinion that canvas2d is legacy and we really shouldn't encourage people to use it, but I dont think like that. Canvas2d will always be here and its needed for better antialiasing, we cant drop the support.

Reply:
That’s fine, this isn’t a blocker for me. However, from my perspective, making something public means we promise to users: we won’t change this, and if we do we won’t break it. Every new public API is a small maintenance tax. Having a very large public API is a big tax. We should be a little more strict about what is public unless there is a very clear reason for public, not just someone MAY use this someday in a plugin.

Reply:
Hm, from that perspective, this thing actually looks good. We can make it part of public API, its formed from all our canvas experience.